### PR TITLE
Add board state

### PR DESCRIPTION
**Pull request by SergiLlamasGuerrero**
As we were talking, I created a sealed class. But I was thinking... Is it necessary? Or maybe an enum class could be enough? What do you think?

### CODE REVIEW

File: app/src/main/java/com/theopensourcefamily/chessclock/BoardState.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.theopensourcefamily.chessclock
+
+sealed class BoardState
```
Comment (joanb) [inline app/src/main/java/com/theopensourcefamily/chessclock/BoardState.kt]: Super unimportant, but as we are not dealing with the physical board in this app, we could choose a name related with the clocks, like ClockState, or CountdownsState, or ChronosState, or Simply MainState (or however we name the activity/fragment + State). What do you think? 🤔

File: app/src/main/java/com/theopensourcefamily/chessclock/BoardState.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+object Stopped : BoardState()
+object WhiteRunning : BoardState()
```
Comment (joanb) [inline app/src/main/java/com/theopensourcefamily/chessclock/BoardState.kt]: I'll answer your PR question regarding the Enum vs Sealed here if you don't mind 😸 
The idea behind this class is to be the only source of truth about what's actually shown in the screen. That means we won't be able to render everything with a state without params, here's why:
Every second (for example while whites are running) a new state is supposed to arrive to be rendered, and that state needs to have the clocks info: whitesCount, blacksCount, ammount of turns, etc... and that parameters need to be part of the data class extending the sealed class.
This atomic states make the view more robust, as you cannot fall in inconsistent states, like running + showing the play button, since it will be only displayed in the pause and stopped states.
So to sum up: everytime the screen needs to change, a new state has to be rendered, and that state has to fyully contain all the necessary info (like a viewmodel). There are some exceptions that are not states (like snackbar, or toasts) that need to be addressed in other ways, but we'll discuss it in another PR! :D)

File: app/src/main/java/com/theopensourcefamily/chessclock/BoardState.kt [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+object BlackRunning : BoardState()
```
Comment (joanb) [inline app/src/main/java/com/theopensourcefamily/chessclock/BoardState.kt]: Hmmm i have doubt about squashing blacks + whites running states, it could be a boolean inside the sealed class, but that would mean treat that comparison inside the view... maybe i like it more the way you did it, what do you think?

Comment (joanb) [reply app/src/main/java/com/theopensourcefamily/chessclock/BoardState.kt]: Ok i've thought about it a little more, and I see your approach definitely better than a single state for both blacks and whites 🚀

File: app/src/main/java/com/theopensourcefamily/chessclock/BoardState.kt (hunk 1 above)
Comment (SergiLlamasGuerrero) [reply app/src/main/java/com/theopensourcefamily/chessclock/BoardState.kt]: Absolutly true! Sorry, I don't know why I named it "board"

File: app/src/main/java/com/theopensourcefamily/chessclock/BoardState.kt (hunk 2 above)
Comment (SergiLlamasGuerrero) [reply app/src/main/java/com/theopensourcefamily/chessclock/BoardState.kt]: Thank you very much for this detailed explanation! You convinced me since second 1 :)

File: app/src/main/java/com/theopensourcefamily/chessclock/BoardState.kt (hunk 1 above)
Comment (joanb) [reply app/src/main/java/com/theopensourcefamily/chessclock/BoardState.kt]: No need to be sorry, we're here to have fun :D!

File: app/src/main/java/com/theopensourcefamily/chessclock/ClockState.kt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package com.theopensourcefamily.chessclock
+
+sealed class ClockState
+object Stopped : ClockState()
+object WhiteRunning : ClockState()
+object BlackRunning : ClockState()
```
Comment (joanb) [inline app/src/main/java/com/theopensourcefamily/chessclock/ClockState.kt]: You can squash the two commits: 
`git reset --soft HEAD~1`
`git commit --amend --no-edit`
`git push --force`

**Comment by joanb** [comment]
@all-contributors please add @SergiLlamasGuerrero  Add first version of states

**Comment by joanb** [comment]
@all-contributors please add @SergiLlamasGuerrero for code

File: app/src/main/java/com/theopensourcefamily/chessclock/ClockState.kt (hunk 4 above)
Comment (SergiLlamasGuerrero) [reply app/src/main/java/com/theopensourcefamily/chessclock/ClockState.kt]: Done!!! :D
